### PR TITLE
[CAL-394] Dropdown refactoring

### PR DESCRIPTION
**Pull request by zomars**
- Prepares Button to pass refs for radix-ui
- Upgrades radix-ui components
- Fixes Dialog crash by upgrading
- Updates Dropdown component with radix-ui

### CODE REVIEW

File: components/ui/Button.tsx [hunk 1]
```diff
@@ -18,7 +18,10 @@ export type ButtonProps = {
   | (JSX.IntrinsicElements["button"] & { href?: never })
 );
 
-export const Button = function Button(props: ButtonProps) {
+export const Button = forwardRef<HTMLAnchorElement | HTMLButtonElement, ButtonProps>(function Button(
+  props: ButtonProps,
+  forwardedRef
+) {
```
Comment (KATT) [inline components/ui/Button.tsx:24]: Nice 👌

File: components/ui/Dropdown.tsx [hunk 2]
```diff
@@ -1,20 +1,92 @@
...
+);
+
+type DropdownMenuItemProps = ComponentProps<typeof DropdownMenuPrimitive["CheckboxItem"]>;
+export const DropdownMenuItem = forwardRef<HTMLDivElement, DropdownMenuItemProps>(
+  ({ className = "", ...props }, forwardedRef) => (
+    <DropdownMenuPrimitive.Item
+      className={`text-sm text-gray-700 hover:bg-gray-100 hover:text-gray-900 ${className}`}
+      {...props}
+      ref={forwardedRef}
+    />
```
Comment (KATT) [inline components/ui/Dropdown.tsx:51]: Minor change (not a blocker) - if you revert order of `className` and spread, you can skip `className`. 

```suggestion
  (props, forwardedRef) => (
    <DropdownMenuPrimitive.Item
      {...props}
      className={classNames(`text-sm text-gray-700 hover:bg-gray-100 hover:text-gray-900 ${className}`, props.className)}
      ref={forwardedRef}
    />
```

**Review by KATT — APPROVED** [review]
Minor comments but LGTM

**Review by pumfleet — APPROVED** [review]

**Review by emrysal — APPROVED** [review]

**Review by PeerRich — APPROVED** [review]

File: components/team/TeamListItem.tsx [hunk 3]
```diff
@@ -124,8 +120,8 @@ export default function TeamListItem(props: {
                       {" "}
                       Edit team
                     </Button>
-                  </li>
-                  <li className="text-sm text-gray-700 hover:bg-gray-100 hover:text-gray-900" role="menuitem">
+                  </DropdownMenuItem>
+                  <DropdownMenuItem className="">
```
[the remark this answers is not in the document]
Comment (zomars) [reply components/team/TeamListItem.tsx:124]: My bad. Nice catch

**Pull request merged**